Utilities/MCStepLogger/include/MCStepLogger/AnalysisFile.h

```diff
@@ -0,0 +1,95 @@
+/* Interface to files produced with the analysis framework
+ * Write files, read them form disk and provide access to histograms and meta information
+ */
+
+#ifndef ANALYSISFILE_H_
+#define ANALYSISFILE_H_
+
+#include <string>
+#include <vector>
+
+#include "TH1.h"
+
+namespace o2
+{
+namespace mcstepanalysis
+{
+
+struct AnalysisMetaInfo {
+  std::string mName;
+  std::string mFramework;  // the framework the MCStepLogger file was produced with
+  std::string mLabel;      // label, also shown in plots, especially comparison plots
+  std::string mVersionTag; // to trace the development of the analysis
+
+  ClassDefNV(AnalysisMetaInfo, 1);
+};
+
+class AnalysisMetaInfo;
+class AnalysisMeasure;
+
+class AnalysisFile
+{
+ public:
+  AnalysisFile();
+  virtual ~AnalysisFile();
+  /// read analysis obects from file
+  bool read(const std::string& filepath);
+  /// write to a directory, analysis name and file name are derived here
+  void write(const std::string& directory);
+  /// delete all member pointers
+  void close();
+  // set data members
+  /// add a single histogram
+  void addHistogram(TH1* histogram, bool isDummy = false);
+  /// set new bunch of histograms
+  void setHistograms(const std::vector<TH1*>& histograms);
+  /// set meta info, warn if already set
+  void setMetaInfo(const std::string& analysisName, const std::string& framework, const std::string& label, const std::string& versionTag);
+  /// get one histogram by name
+  TH1* getHistogram(const std::string& name, bool isDummy = false);
+  /// get all histograms
+  void getHistograms(std::vector<TH1*>& histograms) const;
+  /// retrieve the analysis name
+  const std::string& analysisName() const;
+  /// retrieve the label given during the analysis run
+  const std::string& label() const;
+  /// version tag
+  const std::string& versionTag() const;
+  /// print the meta info
+  void printMetaInfo() const;
+  /// print histogram names
+  void printHistograms() const;
+  /// check whether something has changed, only 'false' if it was only read from file
+  bool hasChanged() const
+  {
+    return mHasChanged;
+  }
+  /// check for and create a directory
+  static bool createDirectory(const std::string& dir);
+
+ public:
+  /// static members which can be used as helpers
+  static const std::string UNKNOWNLABEL;
+
+ private:
+  /// directory name where meta info is stored
+  static const std::string mAnalysisMetaInfoDirName;
+  /// directory name where histograms are stored
+  static const std::string mAnalysisObjectsDirName;
+  /// input file path
+  std::string mInputFilepath;
+  /// meta info
+  AnalysisMetaInfo* mAnalysisMetaInfo;
```

Comment:
why pointer?

Reply:
not any more. Since this class is the interface between the meta info and the outside world making it an object makes totally sense, you are right